sklearn/covariance/_shrunk_covariance.py

```diff
@@ -20,9 +20,36 @@
 from . import empirical_covariance, EmpiricalCovariance
 from .._config import config_context
 from ..utils import check_array
-from ..utils._param_validation import Interval
+from ..utils._param_validation import Interval, validate_params
 
 
+def _oas(X, *, assume_centered=False):
+    """Estimate covariance with the Oracle Approximating Shrinkage algorithm."""
+    # for only one feature, the result is the same whatever the shrinkage
+    if len(X.shape) == 2 and X.shape[1] == 1:
+        if not assume_centered:
+            X = X - X.mean()
+        return np.atleast_2d((X**2).mean()), 0.0
+
+    n_samples, n_features = X.shape
```

Comment:
I removed the following check as this is moved inside the class

::**Removed**::
```
- if X.ndim == 1:
-     n_samples = 1
-     n_features = X.size
- else:
-     n_samples, n_features = X.shape
```

::**Addition**::
```
+     n_samples, n_features = X.shape
```